mofacts/server/methods.js

```diff
@@ -390,26 +390,26 @@ async function setComponentStatesByUserIdTDFIdAndUnitNum(userId, TDFId, componen
       }
 
       const updateQuery = 'UPDATE componentState SET probabilityEstimate=${probabilityEstimate}, \
-          firstSeen=${firstSeen}, lastSeen=${lastSeen}, trialsSinceLastSeen=${trialsSinceLastSeen}, \
-          priorCorrect=${priorCorrect}, priorIncorrect=${priorIncorrect}, \
-          priorStudy=${priorStudy}, totalPracticeDuration=${totalPracticeDuration}, outcomeStack=${outcomeStack} \
-          WHERE userId=${userId} AND TDFId=${TDFId} AND KCId=${KCId} AND componentType=${componentType} \
-          RETURNING componentStateId';
-      try {
-        const componentStateId = await t.one(updateQuery, componentState);
-        resArr.push(componentStateId);
-      } catch (e) {
-        // ComponentState didn't exist before so we'll insert it
-        if (e.name == 'QueryResultError') {
-          const componentStateId = await t.one('INSERT INTO componentState(userId,TDFId,KCId,componentType, \
-            probabilityEstimate,firstSeen,lastSeen,trialsSinceLastSeen,priorCorrect,priorIncorrect,priorStudy, \
-            totalPracticeDuration,outcomeStack) VALUES(${userId},${TDFId}, ${KCId}, ${componentType}, \
-            ${probabilityEstimate},${firstSeen},${lastSeen},${trialsSinceLastSeen},${priorCorrect},${priorIncorrect}, \
-            ${priorStudy},${totalPracticeDuration},${outcomeStack}) \
-            RETURNING componentStateId',
-          componentState);
-        } else {
-          resArr.push('not caught error:', e);
+      hintLevel=${hintLevel}, firstSeen=${firstSeen}, lastSeen=${lastSeen}, trialsSinceLastSeen=${trialsSinceLastSeen}, \
+      priorCorrect=${priorCorrect}, priorIncorrect=${priorIncorrect}, \
+      priorStudy=${priorStudy}, totalPracticeDuration=${totalPracticeDuration}, outcomeStack=${outcomeStack} \
+      WHERE userId=${userId} AND TDFId=${TDFId} AND KCId=${KCId} AND componentType=${componentType} \
```

Comment:
In the read functions you treat hintlevel as part of a composite primary key but in this function you don't.  Shouldn't we be querying based on hint level as well here?